demo/src/execute.ts

```diff
@@ -37,23 +61,35 @@ export class AVRRunner {
     this.usart = new AVRUSART(this.cpu, usart0Config, this.speed);
   }
 
-  async execute(callback: (cpu: CPU) => void) {
-    this.stopped = false;
-    const workUnitCycles = 500000;
-    let nextTick = this.cpu.cycles + workUnitCycles;
+  // CPU main loop
+  execute(callback: (cpu: CPU) => void) {
+    // Initialize on first function start
+    // Not sure why this is useful since this.cpu.cycles is always = to 0 on start
+    // Maybe for the possibility to stop and resume execution ?
+    if (this.nextTick === 0) {
+      this.nextTick = this.cpu.cycles + this.workUnitCycles;
+    }
+
     for (;;) {
```

Comment:
Sync execute is no longer `async()`, I'd replace the infinite loop with a simple for loops that goes while `cpu.cycles < this.nextTick`, and move all the code that calculated the `nextTick` outside the loop. This should simplify the code.

Also, I'd rename `nextTick` to `nextBreak` or `nextPause` or similar, since the method `tick()` already implies that an iteration of the loop is a tick (that's my bad for choosing the same name for two different things)

Reply:
Yeah, i completely missed that. In fact we don't even need this.nextTick anymore since we loop for a fix amount of workUnitCycles.